app/controllers/application_controller.rb

```diff
@@ -70,7 +70,7 @@ def after_sign_in_path_for(resource_or_scope)
   end
 
   def redirect_user_back_after_login
-    if request.env['REQUEST_URI'].present? && !request.xhr? && session[:return_to].nil?
+    if request.env['REQUEST_URI'].present? && !request.xhr?
```

Comment:
@diogob this not break the fallback cases, because fallback is a real route, this session[:return_to].nil? are used when fallback_create was an action.